components/brave_ads/core/internal/history/database_history_util_internal.cc

```diff
@@ -0,0 +1,57 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/core/internal/history/database_history_util_internal.h"
+
+#include <utility>
+
+#include "base/functional/bind.h"
+#include "brave/components/brave_ads/core/internal/account/confirmations/queue/confirmation_queue_database_table.h"
+#include "brave/components/brave_ads/core/internal/common/database/database_transaction_util.h"
+#include "brave/components/brave_ads/core/internal/common/time/time_formatting_util.h"
+#include "brave/components/brave_ads/core/internal/creatives/conversions/creative_set_conversion_database_table.h"
+#include "brave/components/brave_ads/core/internal/settings/settings.h"
+#include "brave/components/brave_ads/core/internal/user_engagement/ad_events/ad_event_cache_util.h"
+#include "brave/components/brave_ads/core/internal/user_engagement/ad_events/ad_events_database_table.h"
+#include "brave/components/brave_ads/core/mojom/brave_ads.mojom.h"
+
+namespace brave_ads {
+
+void MaybeClearDatabaseHistoryForDateRange(const base::Time from_time,
+                                           const base::Time to_time,
+                                           ResultCallback callback) {
+  // Do not clear history for Rewards users from the database, as it is
```

Comment:
Same problem here: we can treat Rewards users differently given that they have opted in, but we need to make it clear whether or not data will be deleted.

Reply:
The concern is fixed by adding `Clear Brave Ads data...` link to the settings for non-Rewards users. 
When clicking this link all Brave Ads files and preferences are deleted.

If user joined Brave Rewards then `Clear Brave Ads data...` link is replaced by `Reset Brave Rewards data...`, which allows to reset rewards and clear all Brave Ads files and preferences as well.